packages/set/src/index.ts

```diff
@@ -88,13 +88,14 @@ export class ReactiveSet<T> extends Set<T> {
   }
 
   clear(): void {
-    if (super.size) {
+    if (!super.size) return;
+    batch(() => {
+      this.#triggers.dirty($KEYS);
+      for (const member of super.values()) {
+        this.#triggers.dirty(member);
+      }
       super.clear();
```

Comment:
I did some benchmarking and did not see a performance hit to moving `super.clear()` into the batch operation. But if you like, I can save a snapshot prior to `.clear()` and iterate over that in batch.